src/SFML/System/FileInputStream.cpp

```diff
@@ -60,13 +60,17 @@ FileInputStream& FileInputStream::operator=(FileInputStream&&) = default;
 
 
 ////////////////////////////////////////////////////////////
-bool FileInputStream::open(const std::string& filename)
+bool FileInputStream::open(const std::filesystem::path& filename)
 {
 #ifdef SFML_SYSTEM_ANDROID
     m_file = std::make_unique<priv::ResourceStream>(filename);
     return m_file->tell() != -1;
+#else
+#ifdef SFML_SYSTEM_WINDOWS
+    m_file.reset(_wfopen(filename.c_str(), L"rb"));
```

Comment:
`std::fopen` does not work on Windows? The cppreference documentation seems to state it is possible.

Reply:
I don't know what the "right" solution is, but handling non ASCII characters is still a major issue and I think `std::fopen` can't handle it properly. Not sure if `c_str()` however defeats this. I could expect a `filename.wstring()` here if we go with `_wfopen`.

Reply:
`std::filesystem::path::c_str()` returns a `const wchar_t*` on Windows which is not convertible to the `const char*` that `std::fopen` needs. This is one instance of a larger problem that Windows paths use wide characters and many external APIs we're using (including `std::fopen`) use `const char*` for paths. You can do a lossy conversion like `path.string().c_str()` which is guaranteed to convert to a `std::string` then a `const char*` but that may break code that actually needs to use non-ASCII characters in paths. 

Because there's apparently a Windows-specific analog to `std::fopen` I added preprocess logic to use that when appropriate but in most other circumstances, the lossy conversion is the best option I could come up with.

Reply:
Right, I checked and unless `std::fstream` is used, one has to work around using macros like this, sadly.

Reply:
I feel like once `std::filesystem` support lands in the API, there will be a whole second effort testing and validating that non-ASCII paths can be used if that's even possible. I know that Freetype and OpenAL are two of the external APIs we're using that expects `const char*` paths so there may be limitation to how widespread we can make non-ASCII support if 3rd party APIs don't support it.